### PR TITLE
Added PUT request and Mapping + Display Phone number on profile page

### LINKED ISSUE #12: create UserProfileController with PUT route to /api/userprofile
# SECOND ISSUE IN #8 

# User Story 

* As a user
* I can add my cell phone
* So that drivers / riders can communicate with one another

### Acceptance Criteria

Allow updating of cell phone in backend

- [ ] There is a new UserProfileController that has a route `PUT /api/userprofile` that takes a single parameter `cellPhone` of type string.  If there is a logged in user, it updates the cell phone number for that logged in user; otherwise it returns a 403.

### LINKED ISSUE #13: Display cell phone number on Profile Page
# THIRD ISSUE IN #8 

# User Story 

* As a user
* I can add my cell phone
* So that drivers / riders can communicate with one another

### Acceptance Criteria

Display cell phone on user profile page

- [ ] On the `/profile` page (i.e. in the `frontend/src/main/pages/ProfilePage.js` component) the cell phone number is displayed.

### PR DESCRIPTION
**Pull request by Aasish-Virjala**
added PUT request mapping in UserProfileController.java to update user cell phone number

Screenshot 2023-08-15 at 5 47 56 PM

Added the ability to view phone number on profile page. Screenshot attached below.

Screenshot 2023-08-15 at 8 18 33 PM

Closes #12 
Closes #13

### CODE REVIEW

File: frontend/src/main/pages/ProfilePage.js [hunk 1]
```diff
@@ -3,9 +3,24 @@ import { Row, Col } from "react-bootstrap";
 import RoleBadge from "main/components/Profile/RoleBadge";
 import { useCurrentUser } from "main/utils/currentUser";
 import BasicLayout from "main/layouts/BasicLayout/BasicLayout";
+// import axios from "axios";
 
 const ProfilePage = () => {
 
+    // Stryker disable all
```
Comment (pconrad) [inline frontend/src/main/pages/ProfilePage.js:10]: Please remove the commented out code

Comment (Aasish-Virjala) [reply frontend/src/main/pages/ProfilePage.js]: This is necessary to remove the testing for Modals as mentioned on Slack. I removed my jank "console.log" strategy as it is not good to have in a "production" product or on the main branch

**Review by pabucsb — COMMENTED** [review]
code looks good. Followed camel case guidelines for cellPhone. Controller and controller test file makes sense.

**Review by alecmorrison — DISMISSED** [review]
Controller and tests look good, frontend screenshot looks good, could have split front end and backend changes into two PRs

**Review by pconrad — CHANGES_REQUESTED** [review]
Please remove the one comment.. it doesn't add any value, since we have github to track changes.

**Review by pconrad — CHANGES_REQUESTED** [review]
Sorry that I didn't notice this before, but there's one more thing to check ..

File: src/main/java/edu/ucsb/cs156/gauchoride/controllers/UserProfileController.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+@Tag(name="Current User Profile")
+@RequestMapping("/api/userprofile")
+@RestController
+public class UserProfileController extends ApiController {
+  @Autowired
+  UserRepository userRepository;
+  @Operation(summary = "Update user cell phone number")
+  @PreAuthorize("hasRole('ROLE_USER')")
+  @PutMapping("")
+  public User updateRide(
```
Comment (pconrad) [inline src/main/java/edu/ucsb/cs156/gauchoride/controllers/UserProfileController.java:31]:
```suggestion
  public User updateUsersCellPhone(
```

**Comment by Aasish-Virjala** [comment]
Fixed the updateRide method to be updateUserCellphone

**Review by pconrad — APPROVED** [review]
LGTM!

**Pull request merged**